nectar/downloaders/threaded.py

```diff
@@ -111,6 +112,58 @@ def progress_interval(self):
         seconds = self.config.progress_interval or DEFAULT_PROGRESS_INTERVAL
         return datetime.timedelta(seconds=seconds)
 
+    @staticmethod
+    def requests_kwargs_from_nectar_config(config):
+        """
+        Take a Nectar configuration and map it to a set of requests keyword arguments.
+
+        These keyword arguments can be used with the Python requests ``requests.request``
+        API. In the future when Nectar is just a memory, this can be adapted to map a
+        Pulp importer configuration to requests kwargs.
+
+        :param config: A nectar configuration instance
+        :type  config: nectar.config.DownloaderConfig
+
+        :return: A dictionary of keyword arguments for the requests API.
+        :rtype:  dict
+        """
+        requests_kwargs = {}
+
+        # Configure basic authentication
+        if config.basic_auth_username and config.basic_auth_password:
+            requests_kwargs['auth'] = (config.basic_auth_username, config.basic_auth_password)
+
+        # Configure verification of the server's TLS certificates; defaults to the system trust store.
+        if config.ssl_validation is not False:
+            if config.ssl_ca_cert_path:
+                requests_kwargs['verify'] = config.ssl_ca_cert_path
+            else:
+                requests_kwargs['verify'] = True
+        else:
+            requests_kwargs['verify'] = False
+
+        # Configure client-side certificate authentication
+        if config.ssl_client_cert_path and config.ssl_client_key_path:
+            requests_kwargs['cert'] = (config.ssl_client_cert_path, config.ssl_client_key_path)
+
+        # Configure proxy servers and proxy authentication.
+        #
+        # Annoyingly, although the config is called 'proxy_url', the port and basic auth
+        # credentials are defined separately, so we have to build the url.
+        if config.proxy_url and config.proxy_port:
```

Comment:
You might notice this does not match the old ``_add_proxy`` method. The bug that method was working around was fixed in requests 2.1 or something. RHEL6 ships requests-2.6 and I confirmed the bug was not present in 2.6.